records_mover/records/schema/field/pandas.py

```diff
@@ -1,15 +1,53 @@
+import pandas as pd
 from pandas import Series, Index
-from typing import Any, Type, TYPE_CHECKING
+from typing import Any, Type, TYPE_CHECKING, Optional
 from .statistics import RecordsSchemaFieldStringStatistics
 from ...processing_instructions import ProcessingInstructions
 from .representation import RecordsSchemaFieldRepresentation
+from ....utils.limits import (INT8_MIN, INT8_MAX,
+                              UINT8_MIN, UINT8_MAX,
+                              INT16_MIN, INT16_MAX,
+                              UINT16_MIN, UINT16_MAX,
+                              INT32_MIN, INT32_MAX,
+                              UINT32_MIN, UINT32_MAX,
+                              INT64_MIN, INT64_MAX,
+                              UINT64_MIN, UINT64_MAX)
 from .numpy import details_from_numpy_dtype
 import numpy as np
 if TYPE_CHECKING:
     from ..field import RecordsSchemaField  # noqa
     from ..schema import RecordsSchema  # noqa
 
 
+def supports_nullable_ints() -> bool:
+    """Detects if this version of pandas supports nullable int extension types."""
+    return 'Int64Dtype' in dir(pd)
+
+
+# Min, max, numpy traditional dtype, pandas extension dtype
+INT_DTYPE_ASSIGNMENT = [
+    (INT8_MIN, INT8_MAX, 'int8', 'Int8'),
+    (UINT8_MIN, UINT8_MAX, 'uint8', 'UInt8'),
+    (INT16_MIN, INT16_MAX, 'int16', 'Int16'),
+    (UINT16_MIN, UINT16_MAX, 'uint16', 'UInt16'),
+    (INT32_MIN, INT32_MAX, 'int32', 'Int32'),
+    (UINT32_MIN, UINT32_MAX, 'uint32', 'UInt32'),
+    (INT64_MIN, INT64_MAX, 'int64', 'Int64'),
+    (UINT64_MIN, UINT64_MAX, 'uint64', 'UInt64'),
+]
```

Comment:
This is feeling like it may be evolving into a class / enum or the like.

Reply:
Yeah, I spent a ridiculous amount of time on writing and rewriting this style-wise, and searching for public numpy or pandas functions related to types. And then I finally gave up and did this. I didn't feel good but it got the job done without the branching complexity that flake8 complained about.

I feel like I want to see how other schema->pandas conversion stuff goes, see what patterns arise, and then revisit this.

Reply:
Well, I went for it.